docs/configuration.asciidoc

```diff
@@ -15,20 +13,319 @@ server_url: 'http://localhost:8200'
+In less regexy terms: Your service name must only contain characters from the ASCII
+alphabet, numbers, dashes, underscores and spaces.
+
+[float]
+[[config-service-version]]
+==== `service_version`
+[options="header"]
+|============
+| Environment                    | `Config` key      | Default   | Example
+| `ELASTIC_APM_SERVICE_VERSION`  | `service_version` | `git` sha | A string indicating the version of the deployed service
+|============
+
+Deployed version of your service.
+Defaults to `git rev-parse --verify HEAD`.
+
+[float]
+[[config-environment]]
+==== `environment`
+
+[options="header"]
+|============
+| Environment               | `Config` key   | Default    | Example
+| `ELASTIC_APM_ENVIRONMENT` | `environment`  | From `ENV` | `"production"`
+|============
+
+The name of the environment this service is deployed in,
+e.g. "production" or "staging".
+
+Defaults to `ENV['RAILS_ENV'] || ENV['RACK_ENV']`.
+
+[float]
+[[config-enabled-environments]]
+==== `enabled-environments`
+
+[options="header"]
+|============
+| Environment                        | `Config` key           | Default
+| `ELASTIC_APM_ENABLED_ENVIRONMENTS` | `enabled_environments` | `['production']`
+|============
+
+Specify which environments to enable APM in.
+
+If the current envrironment isn't included, the agent will effectively be a _noop_
+and do nothing.
+So you can keep in your custom instrumentation code without the agent sending
+anything nor complain.
+
+[float]
+[[config-framework-name]]
+==== `framework_name`
+[options="header"]
+|============
+| Environment                  | `Config` key     | Default
+| `ELASTIC_APM_FRAMEWORK_NAME` | `framework_name` | Depending on framework
+|============
+
+Name of the used framework.
+For Rails or Sinatra, this defaults to `Ruby on Rails` and `Sinatra` respectively,
+otherwise defaults to `nil`.
+
+[float]
+[[config-framework-version]]
+==== `framework_version`
+[options="header"]
+|============
+| Environment                     | `Config` key        | Default
+| `ELASTIC_APM_FRAMEWORK_VERSION` | `framework_version` | Depending on framework
+|============
+
+Version number of the used framework.
+For Ruby on Rails and Sinatra, this defaults to the used version of the framework,
+otherwise, the default is `nil`.
+
+[float]
+[[config-hostname]]
+==== `hostname`
+
+[options="header"]
+|============
+| Environment                | `Config` key  | Default    | Example
+| `ELASTIC_APM_HOSTNAME`     | `hostname`    | `hostname` | `app-server01.example.com`
+|============
+
+The host name to use when sending error and transaction data to the APM server.
+
+[float]
+[[config-source-lines-error-app-frames]]
+==== `source_lines_error_app_frames`
+[float]
+[[config-source-lines-error-library-frames]]
+==== `source_lines_error_library_frames`
+[float]
+[[config-source-lines-span-app-frames]]
+==== `source_lines_span_app_frames`
+[float]
+[[config-source-lines-span-library-frames]]
+==== `source_lines_span_library_frames`
```

Comment:
How about `Number of stacktrace frames to collect` or something similar instead of having 4 headers here?